log.go

```diff
@@ -142,6 +146,25 @@ func (l *logger) log(level Level, msg interface{}, keyvals ...interface{}) {
 	_, _ = l.w.Write(l.b.Bytes())
 }
 
+// readLevelFromEnv reads log level from environment variables if exists.
+func readLevelFromEnv() Level {
```

Comment:
I think it would be more flexible to have a `LoggerOption` that parses from a string, and then the caller can choose to parse it from an `os.Getenv`, a flag, or anything else... 

that would also allow to use different variable names..

wdyt?

Reply:
I think it would be better than this approach. For example, I can add a function called `WithLevelFromEnvVar(key string)` and we will be able to read the log level with a custom key from env variable. It makes it more flexible. But, I didn't get your point in a flag. 

> a flag, or anything else... 

What do you mean by here?

Reply:
I mean, an option like `WithLevelFromString(lvl string)`, then the caller can choose from where it comes from... can be a flag (e.g. `--log.level debug` or from some configuration file or anything else... 

in the case of env vars, callers can `WithLevelFromString(os.Getenv("LOG_LEVEL"))`, for example

Reply:
Perhaps a `ParseLevel(string) Level` that defaults to `InfoLevel` so it's more generic

Reply:
yeah, sounds good too

Reply:
I implemented `WithLevelFromString()` option and added something to test. To be honest, `TestSubLogger` seems not good. I can write more comprehensive and clean tests in this pull request if you want.

Reply:
More tests are always welcome!

Reply:
Maybe I can create another branch and PR for improving tests after merging this PR. What do you think?

Reply:
Even better

Reply:
Sorry for requesting-removing cycle back-to-back 😞 it was a mistake 😥